python/tank_vendor/shotgun_deploy/configuration.py

```diff
@@ -610,19 +596,19 @@ def _update_roots_file(self):
         # now write out roots data
         roots_data = {}
 
-        for storage in self._descriptor.get_required_storages():
-            roots_data[storage] = {}
-            if storage not in storage_by_name:
+        for storage_name in self._descriptor.get_required_storages():
+
+            if storage_name not in storage_by_name:
                 raise ShotgunDeployError(
                     "A '%s' storage is defined by %s but is "
-                    "not defined in Shotgun." % (storage, self._descriptor)
+                    "not defined in Shotgun." % (storage_name, self._descriptor)
                 )
-            roots_data[storage]["mac_path"] = storage_by_name[storage]["mac_path"]
-            roots_data[storage]["linux_path"] = storage_by_name[storage]["linux_path"]
-            roots_data[storage]["windows_path"] = storage_by_name[storage]["windows_path"]
+
+            storage_path = ShotgunPath.from_shotgun_dict(storage_by_name[storage])
```

Comment:
Are you doing from and then to only to cleanup the dict of only valid values? Might as well add a ShotgunPath.clean_shotgun_dict class method?